pkg/services/sqlstore/alert_notification_test.go

```diff
@@ -321,5 +321,54 @@ func TestAlertNotificationSQLAccess(t *testing.T) {
 				So(len(query.Result), ShouldEqual, 4)
 			})
 		})
+
+		Convey("Notification Uid by Id Caching", func() {
+			ss.CacheService.Flush()
+
+			notification := &models.CreateAlertNotificationCommand{Uid: "aNotificationUid", OrgId: 1, Name: "aNotificationUid"}
```

Comment:
@papagian I'm a bit confused.

In the real implementation, [`getAlertNotificationUidInternal`](https://github.com/grafana/grafana/blob/7fb6d2c1d77d08297ab6ab75e5b5a1688826464d/pkg/services/sqlstore/alert_notification.go#L231) is only called if the key is not found in cache and goes to the DB to search for a corresponding UID. If it's not in the DB it returns an error.  
Therefore, it seems redundant to count the method calls, because the test `Retrieves from cache when exists` creates a scenario where a single call to `getAlertNotificationUidInternal` would yield an error. Thus, verifying that it's not called when the item is in cache. 

The corresponding test `Can cache notification Uid` verifies that `getAlertNotificationUidInternal` has been called because the cache is populated with items from the DB. 

So, don't the two tests together already verify these scenarios? 

* I admit I'm a bit biased towards Chicago school TDD 😇

Reply:
Both `Can cache notification Uid` and `Retrieves from cache when exists` actually test that the key is stored in the cache but this does not necessarily mean that the database is not accessed.
Anyway, I don't insist if you think that these scenarios are too exaggerated.

Reply:
I just committed a change that pushes the error down the call path, so that empty query results would return an error and avoid cache population with empty Uid strings.  
So now we can rely on these tests to verify that there really is no DB access when a cached value is found. :)